src/api/spec/controllers/webui/download_on_demand_controller_spec.rb

```diff
@@ -0,0 +1,165 @@
+require 'rails_helper'
+
+RSpec.describe Webui::DownloadOnDemandController do
+  let(:admin_user) { create(:admin_user) }
+  let(:repository) { create(:repository) }
+  let(:project)    { create(:project) }
+
+  let(:dod_parameters) {
+    {
+      project:             project.name,
+      download_repository: {
+        arch:                 "x86_64",
+        repotype:             "rpmmd",
+        url:                  "http://mola.org",
+        archfilter:           "i586",
+        masterurl:            "http://opensuse.org",
+        mastersslfingerprint: "asdfasd",
+        pubkey:               "3jnlkdsjfoisdjf0932juro2ikjfdslñkfj",
+        repository_id:        repository.id
+      }
+    }
+  }
+
+  before do
+    project.repositories << repository
+  end
+
+  it "uses strong parameters" do
+    login(admin_user)
+    should permit(:arch, :repotype, :url, :repository_id, :archfilter, :masterurl, :mastersslfingerprint, :pubkey).
+              for(:create, params: dod_parameters).on(:download_repository)
+  end
+
+  describe "POST create" do
```

Comment:
I agree with both of you. We can introduce a Policy Spec together but in a separate PR